modules/gateway/peers.go

```diff
@@ -273,16 +273,14 @@ func (g *Gateway) Connect(addr modules.NetAddress) error {
 // Disconnect terminates a connection to a peer and removes it from the
 // Gateway's peer list. The peer's address remains in the node list.
 func (g *Gateway) Disconnect(addr modules.NetAddress) error {
-	id := g.mu.RLock()
+	id := g.mu.Lock()
+	defer g.mu.Unlock(id)
```

Comment:
`p.sess.Close()` is a method on the `muxado.Session` type and shouldn't mutate the internal state of the gateway, right?